common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/MicrosoftAuthorizationRequest.java

```diff
@@ -178,6 +184,29 @@ protected MicrosoftAuthorizationRequest(@SuppressWarnings(WarningType.rawtype_wa
         mWorkProfileAvailable = Device.isInPersonalProfileButClouddpcWorkProfileAvailable();
     }
 
+    @Override
+    public URI getAuthorizationRequestAsHttpRequest() throws ClientException {
+        final URI superRequestUri = super.getAuthorizationRequestAsHttpRequest();
+
+        // If the flight is enabled, add fields for AM API Work Profile
+        if (CommonFlightsManager.INSTANCE.getFlightsProvider().isFlightEnabled(CommonFlight.ENABLE_AM_API_WORKPROFILE_EXTRA_QUERY_PARAMETERS)) {
+
+            final CommonURIBuilder builder = new CommonURIBuilder(superRequestUri);
+            builder.addParameterIfAbsent(Device.PlatformIdParameters.MANUFACTURER, mDiagnosticMN);
```

Comment:
Where is `mDiagnosticMN` initialized?

Reply:
Can we just not initialize that if flight is OFF?

Reply:
`mDiagnosticMN = Device.getManufacturer()`, it takes a value from the Device Static instance, which pulls it from device metadata.

I don't really see a benefit to not initializing this field if the flight is off. We aren't doing any additional computation, just fetching a static variable that's already set.

Reply:
The idea is that right now you are using the flight in a hacky way where you are setting these params on AuthorizationRequest regardless of flight status but then only refraining from adding it to the query params if the flight is OFF but I think the better approach would be to not touch the query params and rather check flight before creating and setting these on AuthorizationRequest object

Reply:
That way you don't have to mark them as transient

Reply:
Had an offline discussion, will switch to using the same approach as switch browser field